controllers/infrastructure/byomachine_controller.go

```diff
@@ -260,19 +261,20 @@ func (r *ByoMachineReconciler) reconcileNormal(ctx context.Context, machineScope
 
 	if machineScope.ByoMachine.Spec.ProviderID == "" {
 		logger.Info("Updating Node with ProviderID")
-		providerID := fmt.Sprintf("%s%s/%s", providerIDPrefix, machineScope.ByoHost.Name, util.RandomString(providerIDSuffixLength))
+
 		remoteClient, err := r.getRemoteClient(ctx, machineScope.ByoMachine)
 		if err != nil {
 			logger.Error(err, "failed to get remote client")
 			return ctrl.Result{}, err
 		}
 
-		err = r.setNodeProviderID(ctx, remoteClient, machineScope.ByoHost, providerID)
+		providerID, err := r.setNodeProviderID(ctx, remoteClient, machineScope.ByoHost)
```

Comment:
Trying to keep the change minimal. Any reason why `providerID` computation is moved inside `setNodeProviderID`?

Reply:
That's because, providerID may don't need computation, it may return by setNodeProviderID function under some situation.

Reply:
Ah okay, right. Thanks!